### DISCUSSION TITLE
Optimized toolbar

### DISCUSSION THREAD

**Discussion by aTanguay** [BlenderDefender/SuperEasyAnalytics#4] [Ideas]
This is a great addon...I've been thinking about something like this for years! But I have very very limited programming knowledge. 

I'd **LOVE** to see the option to create a sort of "suggested-toolsbar".

When you're busy working on a project, it's hard to break your flow to mess with menus or learn/change hotkeys. So a lot of people don't bother, when they could be saving a LOT of time. I've always wanted something that watches how *I* work - however un-optimized it might be - and then puts together a toolbar of 'greatest hits'.

Or maybe something that at the very least suggests hotkeys that aren't being used that one should learn, and again, at best, puts together some kind of toolbar that puts your most used stuff right up front. It always seemed like something that would be super helpful when it comes to complex programs. Most people probably use 10-15% of the tools in a program like Blender...maybe less! Why not push that stuff up to the top so you can't miss it?

_Anyway. thanks for making this...it's cool. I'm going to try and do what I'm talking about here manually, hopefully using the info I get from the addon._

**Comment by BlenderDefender** [comment]
Thank you for your feedback. Pablo Vazquez gave some similar ideas about this addon in one of his streams. I'm planning to implement something like this in the future, but I guess I have to get myself familiar with MSGBUS for this. In theory, every button, every value, every viewport, every mode can be analyzed. The problem: Somehow, I have to listen to what is going on in Blender and differentiate the events to count them. If you know someone who can help with this, please let me know.

I'm working on it, but please don't expect visible changes within the next few months, I'll inform you about an Update via my Gumroad Mailing List.

Greetings,

[Blender ](http://bit.ly/BDHOMEPAGE) [Defender](https://bit.ly/BDGUMROADD)

**Comment by aTanguay** [comment]
That sounds great. I'll see if there's anyone I know who might know about that. 

I know that its really easy for me to spout ideas, but what it takes to get that done - I'm sure - can be really tough. I just wanted to throw out an idea that's been a dream of mine for YEARS. I think honestly every complicated program could use something like this. So many programs have literally hundreds of features, but customizing your interface takes away flow. I always thought finding these kinds of patterns in usage/behavior would be something that AI would be great for. Once again, just one person spouting off ideas.

Anyway, thanks again for listening, and thanks for your hard work.

**Comment by BlenderDefender** [comment]
So, I've thought about this idea again and I've come to the conclusion, that the Addon would have to collect too much data to realize this feature. This would probably raise mistrust and hurt my reputation. Therefore, I can't implement this feature.

Greetings,
Blender Defender

**Comment by aTanguay** [comment]
Ah, bummer. It would definitely have to collect a lot of data. That's always the trade-off with things like this. Is it worth it to the user in the end.

**Comment by BlenderDefender** [comment]
Yeah, I know that that's a bit sad. We have discussed some other ideas for the addon over on discord though: https://bit.ly/discord-sea
